### PR TITLE
Error on hardware create with Operation system #1638

### PR DESCRIPTION
**Pull request by ramkishor-ch**
Hi @allmightyspiff,

Please **review** this **pull** **request**.

**Fixes**: #1638 

**Title:** 
Error on hardware create with Operation system #1638

**Description:**
Previously with reference code for os parameter different error message was displayed, so that issue has been fixed in this pull request.

Please find the screen shot.

**Screen Shot:** 
Screenshot 2023-06-19 at 11 24 19 PM

Thanks,
Ramkishor Chaladi.

### CODE REVIEW

**Comment by allmightyspiff** [comment]
>\>       self.assertIsInstance(result.exception, ex.SoftLayerError)
>E       AssertionError: IndexError('list index out of range') is not an instance of <class 'SoftLayer.exceptions.SoftLayerError'>

The test case is providing the wrong value for softwareDescription in its Mock object. It should be a dictionary, NOT an array. That is because [SoftLayer_Product_Item.softwareDescrption](https://sldn.softlayer.com/reference/datatypes/SoftLayer_Product_Item/#softwareDescription) is a single object, not an array.

Anyway to fix that error just do the following.
```
$> git diff tests/managers/ordering_tests.py
diff --git a/tests/managers/ordering_tests.py b/tests/managers/ordering_tests.py
index 7e3c8439..fc8b9d93 100644
--- a/tests/managers/ordering_tests.py
+++ b/tests/managers/ordering_tests.py
@@ -344,8 +344,7 @@ def test_get_price_id_list_item_not_found(self):
         price1 = {'id': 1234, 'locationGroupId': '', 'categories': [category1]}
         softwareDescription1 = {'id': 1234, 'longDescription': 'ABCD 1.2-34', 'referenceCode': 'ABCD_9_32'}
         item1 = {'id': 1111, 'keyName': 'ITEM1', 'itemCategory': category1,
-                 'prices': [price1], 'softwareDescription': [softwareDescription1], }
-
+                 'prices': [price1], 'softwareDescription': softwareDescription1, }
```

File: SoftLayer/managers/ordering.py [hunk 1]
```diff
@@ -370,22 +372,28 @@ def get_price_id_list(self, package_keyname, item_keynames, core=None):
                   keynames in the given package
 
         """
-        mask = 'id, description, capacity, itemCategory, keyName, prices[categories]'
+        mask = 'id, description, capacity, itemCategory, keyName, prices[categories], \
+                softwareDescription[id,referenceCode,longDescription]'
```
Comment (allmightyspiff) [inline SoftLayer/managers/ordering.py:376]: Doing the following will remove the extra whitespace from this mask, and should generally fix the other unit tests you have. 
```suggestion
        mask = 'id, description, capacity, itemCategory, keyName, prices[categories], ' \
                'softwareDescription[id,referenceCode,longDescription]'
```

These ones anyway.
```python
>           raise AssertionError(_error_message()) from cause
E           AssertionError: expected call not found.
E           Expected: list_items('PACKAGE_KEYNAME', mask='id, description, capacity, itemCategory, keyName, prices[categories], softwareDescription[id,referenceCode,longDescription]')
E           Actual: list_items('PACKAGE_KEYNAME', mask='id, description, capacity, itemCategory, keyName, prices[categories],                 softwareDescription[id,referenceCode,longDescription]')
```

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

File: SoftLayer/managers/ordering.py [hunk 2]
```diff
@@ -370,22 +372,28 @@ def get_price_id_list(self, package_keyname, item_keynames, core=None):
...
         item_capacity = self.get_item_capacity(items, item_keynames)
 
         prices = []
         category_dict = {"gpu0": -1, "pcie_slot0": -1}
 
         for item_keyname in item_keynames:
+            matching_item = []
+            # Need to find the item in the package that has a matching
+            # keyName with the current item we are searching for
             try:
```
[the remark this answers is not in the document]
Comment (allmightyspiff) [reply SoftLayer/managers/ordering.py:387]:
```diff
diff --git a/SoftLayer/managers/ordering.py b/SoftLayer/managers/ordering.py
index 8b4fabe7..84cb6f4d 100644
--- a/SoftLayer/managers/ordering.py
+++ b/SoftLayer/managers/ordering.py
@@ -372,8 +372,8 @@ def get_price_id_list(self, package_keyname, item_keynames, core=None):
                   keynames in the given package

         """
-        mask = 'id, description, capacity, itemCategory, keyName, prices[categories], \
-                softwareDescription[id,referenceCode,longDescription]'
+        mask = 'id, description, capacity, itemCategory, keyName, prices[categories], ' \
+               'softwareDescription[id,referenceCode,longDescription]'
         items = self.list_items(package_keyname, mask=mask)
         item_capacity = self.get_item_capacity(items, item_keynames)

@@ -384,15 +384,15 @@ def get_price_id_list(self, package_keyname, item_keynames, core=None):
             matching_item = []
             # Need to find the item in the package that has a matching
             # keyName with the current item we are searching for
-            try:
-                for i in items:
-                    referenceCode = utils.lookup(i, 'softwareDescription', 'referenceCode')
-                    if i['keyName'] == item_keyname or referenceCode == item_keyname:
-                        matching_item.append(i)
-            except IndexError as ex:
-                if len(matching_item) == 0:
-                    message = f"Item {item_keyname} does not exist for package {package_keyname}"
-                    raise exceptions.SoftLayerError(message) from ex
+
+            for i in items:
+                referenceCode = utils.lookup(i, 'softwareDescription', 'referenceCode')
+                if i['keyName'] == item_keyname or referenceCode == item_keyname:
+                    matching_item.append(i)
+
+            if len(matching_item) == 0:
+                message = f"Item {item_keyname} does not exist for package {package_keyname}"
+                raise exceptions.SoftLayerError(message)
             matching_item = matching_item[0]

             # we want to get the price ID that has no location attached to it,
```

**Comment by ramkishor-ch** [comment]
Hi @allmightyspiff ,

As suggested, updated the code changes.

Please re-review this pull request.

Thanks,
Ramkishor Chaladi.

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**